modules/twinkleblock.js

```diff
@@ -1287,7 +1291,7 @@ Twinkle.block.callback.update_form = function twinkleblockCallbackUpdateForm(e,
 	data.hardblock = data.hardblock !== undefined ? data.hardblock : false;
 
 	// disable autoblock if blocking a bot
-	if (Twinkle.block.isRegistered && relevantUserName.search(/bot\b/i) > 0) {
+	if (Twinkle.block.userIsBot) {
```

Comment:
Conversation stalled @siddharthvp @QEDK , but am I right that we arrived at just doing the following:

```suggestion
	if (Twinkle.block.userIsBot || relevantUserName.search(/bot\b/i) > 0) {
```

Do we need the `isRegistered` check/early fail?

```suggestion
	if (Twinkle.block.isRegistered && (Twinkle.block.userIsBot || relevantUserName.search(/bot\b/i) > 0)) {
```

Reply:
The isRegistered check is unnecessary as `relevanUserName` is an IP address in that case, which can't match the regex.

Reply:
I think the thinking was to avoid the regex check for IP addresses, it's what I'd do but I don't think it's much of a performance overhead. 🤔

Reply:
That was basically it, yeah; I agree it's not really worth it, just covering bases.